### PR TITLE
Feature/NUI-5771 fill in any automated testing gaps for gauge

### PR DESCRIPTION
**Pull request by Blakesalot**

### CODE REVIEW

File: packages/charts/src/gauge/gauge-util.ts [hunk 1]
```diff
@@ -97,22 +97,14 @@ export class GaugeUtil {
...
-                // apply label clearances
-                const labelClearanceConfig = gaugeConfig.labels?.clearance;
-                const clearance = !isNil(labelClearanceConfig) ? labelClearanceConfig : DONUT_GAUGE_LABEL_CLEARANCE_DEFAULT;
-                grid.config().dimension.margin = {
-                    top: clearance,
-                    right: clearance,
-                    bottom: clearance,
-                    left: clearance,
-                } as IAllAround<number>;
+                gridConfig.dimension.margin = GaugeUtil.getMarginForLabelClearance(gaugeConfig, mode, gridConfig.dimension.margin);
```
Comment (Blakesalot) [inline packages/charts/src/gauge/gauge-util.ts:107]: guerrilla refactor: extracted to getMarginForLabelClearance function below

File: packages/charts/src/gauge/gauge-util.ts [hunk 2]
```diff
@@ -121,17 +113,7 @@ export class GaugeUtil {
...
-            let marginToAdjust: keyof IAllAround<number>;
-            if (mode === GaugeMode.Horizontal) {
-                marginToAdjust = flippedLabels ? "top" : "bottom";
-            } else {
-                marginToAdjust = flippedLabels ? "left" : "right";
-            }
-
-            const clearance = gaugeConfig.labels?.clearance ?? LINEAR_GAUGE_LABEL_CLEARANCE_DEFAULTS[marginToAdjust];
-            grid.config().dimension.margin[marginToAdjust] = clearance;
+            gridConfig.dimension.margin = GaugeUtil.getMarginForLabelClearance(gaugeConfig, mode, gridConfig.dimension.margin);
```
Comment (Blakesalot) [inline packages/charts/src/gauge/gauge-util.ts:116]: guerrilla refactor: extracted to getMarginForLabelClearance function below

File: packages/charts/src/gauge/gauge-util.ts [hunk 3]
```diff
@@ -209,6 +191,79 @@ export class GaugeUtil {
...
 
+    /**
+     * Convenience function for creating a standard standard set of threshold configs. Includes configurations for warning and error thresholds.
+     *
+     * @param warningVal Value for the warning threshold
+     * @param criticalVal Value for the critical threshold
+     *
+     * @returns {IGaugeThresholdsConfig} The thresholds configuration
+     */
+    public static createStandardThresholdsConfig(warningVal: number, criticalVal: number): IGaugeThresholdsConfig {
```
Comment (Blakesalot) [inline packages/charts/src/gauge/gauge-util.ts:202]: moved this function from below

File: packages/charts/src/gauge/gauge-util.ts [hunk 4]
```diff
@@ -209,6 +191,79 @@ export class GaugeUtil {
...
+     * Gets a new instance of the provided grid margin with updated values (if needed) to accommodate the label clearance specified
+     * in the provided gauge configuration
+     *
+     * @param gaugeConfig The gauge's configuration
+     * @param mode The mode of the gauge
+     * @param margin The margin to update
+     *
+     * @returns The updated margin
+     */
+    public static getMarginForLabelClearance(gaugeConfig: IGaugeConfig, mode: GaugeMode, margin: IAllAround<number>): IAllAround<number> {
```
Comment (Blakesalot) [inline packages/charts/src/gauge/gauge-util.ts:237]: new utility function

**Review by elenates — APPROVED** [review]

**Review by baggiosw — APPROVED** [review]

**Pull request merged**